server/src/main/java/org/elasticsearch/plugins/PluginsService.java

```diff
@@ -589,7 +608,7 @@ private String signatureMessage(final Class<? extends Plugin> clazz) {
 
     @SuppressWarnings("unchecked")
     public <T> List<T> filterPlugins(Class<T> type) {
-        return plugins.stream().filter(x -> type.isAssignableFrom(x.v2().getClass()))
-            .map(p -> ((T)p.v2())).collect(Collectors.toList());
+        return reversePlugins.stream().filter(x -> type.isAssignableFrom(x.getClass()))
```

Comment:
Using plugins in reverse ensures that `createComponents` of extending plugins are called before `createComponents` of their `ExtensiblePlugin`, allowing the `ExtensiblePlugin` to collect information immediately in `createComponents`. Many plugins use `createComponents` as an initialization, initializing a bunch of fields based on the input and the components created, necessitating this order.